trio/_threads.py

```diff
@@ -168,6 +179,10 @@ def do_release_then_return_result():
 
     current_trio_token = trio.lowlevel.current_trio_token()
 
+    if thread_name is None:
+        # TODO, better default since it caps at 15 chars
+        thread_name = f"Thread for {trio.lowlevel.current_task().name}"
```

Comment:
Yeah I'm not sure how to default it, 15 chars really isn't a lot and can easily get hit just with the function name. Could do `f'{trio.lowlevel.current_task().name.split(".")[-1]'[:14-len("thread_number") {thread number}]`

I think the last part of the name is usually the most interesting part? Or maybe `.__name__` on the passed function. And I think you might want the thread number in case you're launching several identical ones and want to disambiguate them.

Reply:
I find the thread-ID point pretty convincing, so how about we just `thread_name.removeprefix("Thread for ")` before truncating and assigning the OS thread name?  See other comment for details.

Reply:
Oh actually, I think removing the prefix could trip up the end user. Somebody might manually assign "Thread for [a/b/c/d/e]" as their thread name. I'm ditching "Thread for" entirely, will push up a new version soon.